openproblems/tasks/_batch_integration/batch_integration_embed/datasets/immune.py

```diff
@@ -0,0 +1,32 @@
+from .....data.immune_cells import load_immune
+from .....tools.decorators import dataset
+from .....tools.normalize import log_scran_pooling
+
+import scanpy as sc
+
+# from scIB.preprocessing import normalize, hvg_batch
+
+
+@dataset(dataset_name="Immune (by batch)", image="openproblems-r-base")
+def immune_batch(test=False):
+    adata = load_immune(test=test)
+    from_cache = adata.__from_cache__
+    adata.obs["labels"] = adata.obs["final_annotation"]
+
+    adata.layers["counts"] = adata.X
```

Comment:
please add a comment that `adata.obs['batch']` is already present in the object.

Reply:
and remove all non-relevant `.obs` columns.